packages/office-ui-fabric-react/src/components/SpinButton/SpinButton.tsx

```diff
@@ -44,6 +44,13 @@ export interface ISpinButtonState {
    * The calculated precision for the value.
    */
   precision: number;
+
+  /**
+   * When a typed value is validated and committed by some means other than onBlur, this should be true
+   * to prevent a duplicate submission if onBlur immediately follows the commit. It should be set to false
+   * When the value input changes again, whether by typing or spinning, or when onBlur finally occurs
+   */
+  submissionPending: boolean;
```

Comment:
Does this need to be state (vs a private member) since it doesn't cause any rendering differences?

Reply:
Also, instead of a Boolean, could this potentially be a string that stores the last submitted value? If you did that then you could not call validate if the value did not change

Reply:
The string implementation is analogous to what's done in ComboiBox. But onBlur has the potential to be a redundant validation call (and in our app, an extra executeCommand). I think you're right about it not needing to be state. Let me code that up and see how it looks and works. Good feedback!

Reply:
I think we have something similar for  our pending value in ComboBox. Was there any reason why we should have that as a state too?

---
In reply to: [212616590](https://github.com/OfficeDev/office-ui-fabric-react/pull/6058#discussion_r212616590) [](ancestors = 212616590)